addons/mail/models/mail_thread.py

```diff
@@ -1275,15 +1272,15 @@ def message_route(self, message, message_dict, model=None, thread_id=None, custo
             message_dict.pop('parent_id', None)
 
             # check it does not directly contact catchall
-            if catchall_alias and any(email.startswith(catchall_alias) for email in email_to_localparts):
+            if catchall_alias and all(email_localpart == catchall_alias for email_localpart in email_to_localparts):
```

Comment:
this fixes `test_message_process_write_to_catchall_and_alias` test

Reply:
This fix is good. It is important that a mail containing `To: catchall@example.com, valid-alias@example.com` is not to be rejected because the alias is a good candidate to receive new emails.